### ISSUE TITLE
Cache `context.font` value when rendering

### ISSUE THREAD

**Issue by mdamien** [Linkurious/linkurious.js#168]
Continuing the performance analysis of the edges labels rendering, I found that looking at the `context.font` value is expensive. It's ~10ms over a 40ms render.

See the [bench/edge-labels](https://github.com/Linkurious/linkurious.js/blob/bench/edge-labels) branch. The results on my XPS 13:
 ![results](http://i.imgur.com/rM46zuB.png)

So, a first solution would be for the renderer to cache the value of the latest font.

This performance optimization is only triggered when we don't use save()/restore() although. 

A generic solution is to tranform the `def` function to an object with pre()/post() hooks like:

```
xyz.def = {
   pre: function(){ this.font = "";},
   render: function(..){..use this.font as a context.font cache...},
   post: function(..){..clean things..},
}
```

Think kind of system is kinda already in the svg renderer (via `update()`/`create()` hooks) and the webgl renderer.

So I'm gonna create a new issue to have this system in place and then implement the `context.font` caching.

**Comment by sheymann** [comment]
How to cache the font?

**Comment by mdamien** [comment]
Only the string value of context.font like this: https://github.com/Linkurious/linkurious.js/blob/bench/edge-labels/bench/final.js#L42-L57

**Comment by mdamien** [comment]
Added the perfs of the new renderer included in the #195 PR in the bench branch : https://github.com/linkurious/linkurious.js/commit/94d8797843eda6ddde384bfddaa0119e648bf8a9

![perfs](http://i.imgur.com/b1g7ftA.png)

### Some explanations:

For the edge label renderers,

* `previous` is the one curently in plugin/edge-labels
* `fillText` only do a fillText
* `fillText + angle` do a fillText + ctx.rotate
* `current` is the new renderer

For the node label renderers,

* `current` is the one included in `renderer.linkurious`
* `limit_ctx_switch` do not do any ctx.font access

Next step:

* optimize the node label renderer with the ctx.font caching
* benchmark and optimize if necessary the halo and the glyphs plugins

**Comment by mdamien** [comment]
Now that the ctx.font value is cached in the two most important place (node labels and edges labels), I close this issue and will open new ones for more specific fixes
